Testscripts/Windows/VERIFY-SRIOV-FAILSAFE-XDP.ps1

```diff
@@ -0,0 +1,196 @@
+        Write-LogErr "Test Aborted. Last known status : $currentStatus."
+        Throw "XDP Execution failed"
+    }
+    Write-LogInfo "XDPDump application ran successfully."
+    return "PASS"
+}
+
+function Main {
+    try{
+        $noClient = $true
+		$noServer = $true
+		foreach ($vmData in $allVMData) {
+			if ($vmData.RoleName -imatch "receiver") {
+				$clientVMData = $vmData
+				$noClient = $false
+			}
+			elseif ($vmData.RoleName -imatch "sender") {
+				$noServer = $false
+				$serverVMData = $vmData
+			} else {
+				Write-LogErr "VM role name is not matched with sender or receiver"
+			}
+		}
+		if ($noClient) {
+			Throw "No any receiver VM defined. Aborting Test."
+		}
+		if ($noServer) {
+			Throw "No any sender VM defined. Aborting Test."
+        }
+
+        # CONFIGURE VM Details
+        Write-LogInfo "CLIENT VM details :"
+        Write-LogInfo "  RoleName : $($clientVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($clientVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($clientVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($clientVMData.InternalIP)"
+        Write-LogInfo "SERVER VM details :"
+        Write-LogInfo "  RoleName : $($serverVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($serverVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($serverVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($serverVMData.InternalIP)"
+
+        # Check for compatible kernel
+        $currentKernelVersion = Run-LinuxCmd -ip $clientVMData.PublicIP -port $clientVMData.SSHPort `
+            -username $user -password $password -command "uname -r"
+        if ((Compare-KernelVersion $currentKernelVersion $MIN_KERNEL_VERSION) -lt 0 -or $global:DetectedDistro -ne "UBUNTU"){
+            Write-LogInfo "Unsupported kernel version: $currentKernelVersion or Unsupported distro: $($global:DetectedDistro)."
+            return $global:ResultSkipped
+        }
+
+        # PROVISION VMS
+        Provision-VMsForLisa -allVMData $allVMData -installPackagesOnRoleNames "none"
+
+        # Generate constants.sh and write all VM info into it
+        Write-LogInfo "Generating constants.sh ..."
+        $constantsFile = "$LogDir\constants.sh"
+        Set-Content -Value "# Generated by Azure Automation." -Path $constantsFile
+        Add-Content -Value "ip=$($clientVMData.InternalIP)" -Path $constantsFile
+        Add-Content -Value "nicName=eth1" -Path $constantsFile
+        foreach ($param in $currentTestData.TestParameters.param) {
+            Add-Content -Value "$param" -Path $constantsFile
+        }
+        Write-LogInfo "constants.sh created successfully..."
+        Write-LogInfo (Get-Content -Path $constantsFile)
+
+        # Build and Install XDP Dump application
+        $installXDPCommand = @"
+./XDPDumpSetup.sh 2>&1 > ~/xdpConsoleLogs.txt
+. utils.sh
+collect_VM_properties
+"@
+        Set-Content "$LogDir\StartXDPSetup.sh" $installXDPCommand

```

Comment:
Where is it? files tag does not include this file name.

Reply:
We are creating this file and uploading it. Set-Content will set content in var installXDPCommand into a new file $LogDir\StartXDPSetup.sh